include/oneapi/tbb/detail/_config.h

```diff
@@ -430,6 +430,11 @@
     #define TBB_ALLOCATOR_TRAITS_BROKEN 1
 #endif
 
+// HAIKU build fix
+#if defined(__HAIKU__)
+    #define TBB_ALLOCATOR_TRAITS_BROKEN 1
```

Comment:
Otherwise:
```
In file included from /boot/system/develop/tools/lib/gcc/x86_64-unknown-haiku/8.3.0/include/c++/string:52,
                 from /boot/system/develop/tools/lib/gcc/x86_64-unknown-haiku/8.3.0/include/c++/bits/locale_classes.h:40,
                 from /boot/system/develop/tools/lib/gcc/x86_64-unknown-haiku/8.3.0/include/c++/bits/ios_base.h:41,
                 from /boot/system/develop/tools/lib/gcc/x86_64-unknown-haiku/8.3.0/include/c++/ios:42,
                 from /boot/system/develop/tools/lib/gcc/x86_64-unknown-haiku/8.3.0/include/c++/istream:38,
                 from /boot/system/develop/tools/lib/gcc/x86_64-unknown-haiku/8.3.0/include/c++/fstream:38,
                 from /RAMDisk/oneTBB-2021.1.1/test/common/doctest.h:2739,
                 from /RAMDisk/oneTBB-2021.1.1/test/common/test.h:30,
                 from /RAMDisk/oneTBB-2021.1.1/test/tbb/test_concurrent_hash_map.cpp:23:
/boot/system/develop/tools/lib/gcc/x86_64-unknown-haiku/8.3.0/include/c++/bits/basic_string.h: In instantiation of 'class std::basic_string<char, std::char_traits<char>, tbb::detail::d1::tbb_allocator<char> >':
/RAMDisk/oneTBB-2021.1.1/test/tbb/test_concurrent_hash_map.cpp:365:33:   required from here
/boot/system/develop/tools/lib/gcc/x86_64-unknown-haiku/8.3.0/include/c++/bits/basic_string.h:3106:63: error: no class template named 'rebind' in 'class tbb::detail::d1::tbb_allocator<char>'
       typedef typename _Alloc::template rebind<_CharT>::other _CharT_alloc_type;
                                                               ^~~~~~~~~~~~~~~~~
compilation terminated due to -Wfatal-errors.
test/CMakeFiles/test_concurrent_hash_map.dir/build.make:81: recipe for target 'test/CMakeFiles/test_concurrent_hash_map.dir/tbb/test_concurrent_hash_map.cpp.o' failed
make[2]: *** [test/CMakeFiles/test_concurrent_hash_map.dir/tbb/test_concurrent_hash_map.cpp.o] Error 1
CMakeFiles/Makefile2:1725: recipe for target 'test/CMakeFiles/test_concurrent_hash_map.dir/all' failed
make[1]: *** [test/CMakeFiles/test_concurrent_hash_map.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```